### PR TITLE
view photos

### PR DESCRIPTION
**Pull request by VuManhHung97**
Create UI : https://marvelapp.com/c1cbha4/screen/58854270
Call API and get data
Handle when click item list photos => open PhotoDetail screen

https://edu-redmine.sun-asterisk.vn/issues/14606

photo

### CODE REVIEW

**Comment by VuManhHung97** [comment]
ready

File: app/src/main/java/com/sun_asterisk/myeditor03/screen/photos/PhotosFragment.kt [hunk 1]
```diff
@@ -1,17 +1,70 @@
...
+            override fun onLoadMore() {
+                page++
+                viewModel.getPhotos(page)
+            }
+        })
+    }
+
+    private fun registerLiveData() {
+        viewModel.photoLiveData.observe(this, Observer {
+            if (it != null) {
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/sun_asterisk/myeditor03/screen/photos/PhotosFragment.kt]: check null bên viewModel e

**Comment by VuManhHung97** [comment]
ready

**Comment by VuManhHung97** [comment]
ready

File: app/src/main/java/com/sun_asterisk/myeditor03/utils/BaseAdapter.kt [hunk 2]
```diff
@@ -24,10 +24,12 @@ abstract class BaseAdapter<T> : RecyclerView.Adapter<BaseViewHolder<T>>() {
         notifyItemChanged(data.indexOf(item))
     }
 
-    fun addItems(item: List<T>) {
+    fun addItems(item: List<T>?) {
         val originSize = data.size
-        data.addAll(item)
-        notifyItemRangeInserted(originSize, item.size)
+        if (item != null) {
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/sun_asterisk/myeditor03/utils/BaseAdapter.kt]: e có thể dùng ? or !! trong kotlin để checknull nhé

File: app/src/main/java/com/sun_asterisk/myeditor03/screen/photos/PhotosViewModel.kt [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+class PhotosViewModel(private val photoRepository: PhotoRepository) : ViewModel() {
+    var photoLiveData = MutableLiveData<List<Photo>>()
+    private val compositeDisposable = CompositeDisposable()
+
+    fun getPhotos(page: Int) {
+        compositeDisposable.add(
+            photoRepository.getPhotos(page)
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe({ photoLiveData.value = it }, this::handleError)
```
Comment (tuanlvt-1117) [inline app/src/main/java/com/sun_asterisk/myeditor03/screen/photos/PhotosViewModel.kt]:
```suggestion
                .subscribe({ data -> data.let{ photoLiveData.value = it  } 
                , this::handleError)
```

**Comment by VuManhHung97** [comment]
ready

**Review by tuanlvt-1117 — APPROVED** [review]

**Pull request merged**